File: archive/2020/solutions/EggInterception.py
def brush_key(brush):
    return brush[0]

input_split = input().split()
num_eggs = int(input_split[0])
curr_pos = int(input_split[1]) - 1

brushes = [[0, i] for i in range(num_eggs)]

input_split = input().split()

for i in range(num_eggs):
    brushes[i][0] = int(input_split[i])

brushes.sort(key=brush_key)

move_dist = 0

for brush in brushes:
    move_dist += abs(brush[1] - curr_pos)
    curr_pos = brush[1]

print(move_dist)

File: archive/2020/solutions/EggInterception.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;


struct Brush
{
    long long height;
    long long location;
};

long long currPos;
int numBrushes;
vector<Brush> brushes;

bool brushCmp(Brush b1, Brush b2){
    return b1.height < b2.height;
}

int main(){
    cin >> numBrushes >> currPos;
    currPos--;
    brushes.resize(numBrushes);

    long long currHeight;
    for(int i = 0; i < numBrushes; i++){
        cin >> currHeight;
        brushes[i] = Brush{currHeight - 1, i};
    }

    sort(brushes.begin(), brushes.end(), brushCmp);

    long long moveDist = 0;
    for(Brush b: brushes){
        moveDist += abs(b.location - currPos);
        currPos = b.location;
    }

    cout << moveDist;
}
